pkg/endpoints/credentials.go

```diff
@@ -345,3 +371,82 @@ func credentialToSecret(cred credential, namespace string, response *restful.Res
 	// Return secret
 	return &secret, true
 }
+
+// addSecretToSA - Add secret to service account 
+func (r Resource) addSecretToSA(saName string, secretName string , namespaceName string) bool {
+	sa := corev1.ServiceAccount{
+		Secrets: []corev1.ObjectReference {{Name: secretName},},
+	}
+
+	marshaledSA, err := json.Marshal(sa)
+	if err != nil {
+		logging.Log.Errorf("failed to marshal payload")
```

Comment:
Mention it's failed to `marshal` the payload for the service account

Reply:
I fix it.